src/core/utils/Popover.tsx

```diff
@@ -83,11 +88,26 @@ export const Popover = React.forwardRef((props: PopoverProps, ref) => {
   return <Tippy {...computedProps} ref={refs} />;
 });
 
-export const hideOnEsc = {
+/**
+ * Plugin to hide tippy when either Esc key is pressed,
+ * or when the content inside is not tabbable and Tab key is pressed.
+ */
+export const hideOnEscOrTab = {
   fn(instance: Instance) {
     const onKeyDown = (event: KeyboardEvent) => {
-      if (event.key === 'Escape') {
-        instance.hide();
+      switch (event.key) {
+        case 'Escape':
+          instance.hide();
+          break;
+        case 'Tab':
+          const descendents = Array.from<HTMLElement>(
+            instance.popper.querySelectorAll('*'),
+          );
+          if (!descendents.some((el) => el?.tabIndex >= 0)) {
```

Comment:
This approach is working pretty well on normal `DropdownMenu`, but not so much on `Select`.

Reply:
Why are we setting tabindex=0 on selected menuitem?

https://github.com/iTwin/iTwinUI-react/blob/df6f9fac901ffa771dca4db642e8cdb611551aac/src/core/Menu/MenuItem.tsx#L96

Reply:
Weird behavior I find is that doesnt matter if you Tab or Shift+Tab you always go to the menu focus...

Reply:
Btw, have you noticed that clicking again on the dropdown does not close it in w3? :D
https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html

Reply:
Doesn't work with table filters. I tabbed out and filter is still visible.

Reply:
Tab or Shift+Tab should both go back to button, I made that intentional (using `preventDefault`).

Table filter needs extra logic to keep track of the last focusable element. So I specifically excluded it.

For Select, try selecting an item, reopening the menu, and tabbing out. It won't work because the selected item gets tabindex=0

Reply:
Fixed Tab / Shift+Tab behavior.

For selected item, I will remove tabindex=0 in next PR